scipy/special/basic.py

```diff
@@ -2126,6 +2126,65 @@ def factorialk(n, k, exact=True):
         raise NotImplementedError
 
 
+def voigt(x, sigma=1.0, gamma=1.0, mu=0.0):
+    r"""
+    Voigt profile.
+
+    The Voigt profile is a convolution of a 1D Normal distribution with
+    standard deviation ``sigma`` and a 1D Cauchy distribution with half-width at
+    half-maximum ``gamma``.
+
+    Parameters
+    ----------
+    x : array_like
+        Input argument.
+    sigma : float, optional
+        The standard deviation of the Normal distribution part. The default is
+        1.0.
+    gamma : float, optional
+        The half-width at half-maximum of the Cauchy distribution part. The
+        default is 1.0.
+    mu : float, optional
+        The location of the peak profile. The default is at 0.0.
+
+    Returns
+    -------
+    y : array_like
+        The Voigt profile at the given position. It will have the same shape as
+        `x`.
+
+    Notes
+    -----
+    It can be expressed in terms of Faddeeva function
+
+    .. math:: V(x;\sigma,\gamma,\mu) = \frac{Re[w(z)]}{\sigma\sqrt{2\pi}},
+    .. math:: z = \frac{x-\mu+i\gamma}{\sqrt{2}\sigma}
+
+    where :math:`w(z)` is the Faddeeva function.
+
+    See Also
+    --------
+    wofz : Faddeeva function
+
+    References
+    ----------
+    .. [1] https://en.wikipedia.org/wiki/Voigt_profile
+    """
+    INV_SQRT_2 = 0.7071067811865475
```

Comment:
This isn't necessarily enough digits to roundtrip.

Reply:
> I think M_SQRT1_2 is universally available? Ditto for M_SQRT_2PI

I tried this and included `<math.h>` with and without defining _USE_MATH_DEFINES, and the compilation failed on those cases. So I just take the numbers from [here](https://hahnlab.github.io/CAFE/src_docs/html/mathfunc_8c.html) and [here](https://msdn.microsoft.com/en-us/library/4hwaceh6.aspx)